pkg/noderesourcetopology/plugin.go

```diff
@@ -0,0 +1,74 @@
+package noderesourcetopology
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+	apiconfig "sigs.k8s.io/scheduler-plugins/pkg/apis/config"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+)
+
+// TopologyMatch plugin which run simplified version of TopologyManager's admit handler
+type TopologyMatch struct {
+	nodeResTopologyPlugin
+	policyHandlers      PolicyHandlerMap
+	scoreStrategy       scoreStrategy
+	resourceToWeightMap resourceToWeightMap
+}
+
+var _ framework.FilterPlugin = &TopologyMatch{}
+var _ framework.ScorePlugin = &TopologyMatch{}
+
+const (
+	// Name is the name of the plugin used in the plugin registry and configurations.
+	Name = "NodeResourceTopologyMatch"
+)
+
+// Name returns name of the plugin. It is used in logs, etc.
+func (tm *TopologyMatch) Name() string {
+	return Name
+}
+
+// New initializes a new plugin and returns it.
+func New(args runtime.Object, handle framework.Handle) (framework.Plugin, error) {
+	klog.V(5).Infof("creating new TopologyMatch plugin")
+	tcfg, ok := args.(*apiconfig.NodeResourceTopologyMatchArgs)
+	if !ok {
+		return nil, fmt.Errorf("want args to be of type NodeResourceTopologyMatchArgs, got %T", args)
+	}
+
+	lister, err := getNodeTopologyLister(&tcfg.MasterOverride, &tcfg.KubeConfigPath)
+	if err != nil {
+		return nil, err
+	}
+
+	resToWeightMap := make(resourceToWeightMap)
+	for _, resource := range tcfg.Resources {
+		resToWeightMap[v1.ResourceName(resource.Name)] = resource.Weight
+	}
+
+	// TBD what strategy should be provided by default
+	var scoringStrategy = leastAllocatableScoreStrategy
```

Comment:
This should be defaulted and registered in pkg/apis/config/v1beta1/zz_generated.defaults.go.

Also, we should validate and abort on illegal scoring strategy.